source/transactions-convenient-api/transactions-convenient-api.rst

```diff
@@ -299,6 +299,26 @@ MongoClient within the callback in order to execute operations within the
 transaction. Per the `Driver Session`_ specification, ClientSessions should
 already provide access to a client object.
 
+Handling errors inside the callback
+-----------------------------------
+
+Drivers MUST document that the callback MUST NOT silently handle command errors
+without allowing such errors to propagate. Command errors may abort the transaction
+on the server, and an attempt to commit the transaction will be rejected with
+``NoSuchTransaction`` error.
+
+For example, ``DuplicateKeyError`` is an error that aborts a transaction on the
+server. If the callback catches ``DuplicateKeyError`` and does not re-throw it,
+the driver will attempt to commit the transaction. The server will reject the
+commit attempt with ``NoSuchTransaction`` error. This error has the
+"TransientTransactionError" label and the driver will retry the commit. This
+will result in an infinite loop.
+
+
+Drivers SHOULD recommend that the callback re-throw command errors if they
```

Comment:
Just out of curiosity, why are these a SHOULD and not a MUST? It seems to me like it would be required, 1) to lower the potential cases and 2) to have a quick point of reference for all drivers when users run into this.

Reply:
Sounds reasonable, changed to MUST.